src/Adapter/Product/CommandHandler/UpdateProductDescriptionHandler.php

```diff
@@ -0,0 +1,116 @@
+
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductDescriptionCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\UpdateProductDescriptionHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShopException;
+use Product;
+
+/**
+ * Handles UpdateProductDescriptionCommand using legacy object model
+ */
+final class UpdateProductDescriptionHandler extends AbstractProductHandler implements UpdateProductDescriptionHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateProductDescriptionCommand $command): void
+    {
+        $productId = $command->getProductId();
+        $product = $this->getProduct($productId);
+
+        $localizedDescriptions = $command->getLocalizedDescriptions();
+        $localizedShortDescriptions = $command->getLocalizedShortDescriptions();
+
+        if (null !== $localizedDescriptions) {
+            $this->validateDescriptions($product, $localizedDescriptions, false);
+            $product->description = $localizedDescriptions;
+        }
+
+        if (null !== $localizedShortDescriptions) {
+            $this->validateDescriptions($product, $localizedShortDescriptions, true);
+            $product->description_short = $localizedShortDescriptions;
+        }
+
+        try {
+            if (false === $product->update()) {
+                throw new CannotUpdateProductException(sprintf(
+                    'Error occurred when trying to update product #%s description',
+                    $product->id
+                ));
+            }
+        } catch (PrestaShopException $e) {
+            throw new CannotUpdateProductException(
+                sprintf(
+                    'Error occurred when trying to update product #%s description',
+                    $product->id
+                ),
+                0,
+                $e
+            );
+        }
+    }
+
+    /**
+     * @param Product $product
+     * @param array $localizedDescriptions
+     * @param bool $forShortDescription
+     *
+     * @throws ProductConstraintException
+     * @throws \PrestaShopException
+     */
+    private function validateDescriptions(Product $product, array $localizedDescriptions, bool $forShortDescription): void
+    {
+        $descriptionType = 'description';
+        $errorCode = ProductConstraintException::INVALID_DESCRIPTION;
+
+        if ($forShortDescription) {
+            $descriptionType = 'short description';
+            $errorCode = ProductConstraintException::INVALID_SHORT_DESCRIPTION;
+        }
+
+        foreach ($localizedDescriptions as $langId => $localizedDescription) {
+            if (true !== $product->validateField('description', $localizedDescription, $langId)) {
```

Comment:
```suggestion
            if (true !== $product->validateField($descriptionType, $localizedDescription, $langId)) {
```

And `$descriptionType` should be `description_short` for short descriptions

Reply:
I wondered how the Behat test could work, but it's actually because both fields have the same validation method ^^ Lucky match 😛